packages/spec/assert.savi

```diff
@@ -0,0 +1,41 @@
+:primitive Assert
+  :fun non condition(
+    spec Spec
+    success Bool
+    pos SourceCodePosition = source_code_position_of_argument success
+  )
+    test = spec.test
+    assert = AssertCondition.new(test.spec, test.example, success, pos)
+    test.specs.enqueue(assert)
+
+// TODO: traits don't yet generate a singleton object
+// so we keep `:primitive Assert` as our singleton for the macro's call
+// and use Assert2 to sub-type `AssertCondition` and `AssertRelation`.
+//
+// We'll merge Assert and Assert2 once a `:trait` can be used as a singleton.
```

Comment:
This should now be possible to do.

I believe I've fully patched up this hole in this commit: https://github.com/savi-lang/savi/commit/fe754361d7b59bd866ccdde6df9143f0dcaf09b8

So if you pull latest `main` branch in you should be able to get rid of `Assert2`, either within this PR or in a future PR - whichever works best for you.